Client/SafeBrowsing/GSafeBrowsingDatabase.swift

```diff
@@ -0,0 +1,511 @@
+            completion(results)
+        }
+    }
+    
+    func update(_ fetchResponse: FetchResponse, completion: @escaping (Error?) -> Void) {
+        dbLock.lock(); defer { dbLock.unlock() }
+        
+        self.lastFetchDate = Date()
+        self.lastFetchWaitDuration = Double(fetchResponse.minimumWaitDuration)
+        self.numberOfFetchRetries = 0
+        self.updateDatabaseInfo(completion)
+        
+        fetchResponse.listUpdateResponses.forEach({ response in
+            if response.additions.isEmpty && response.removals.isEmpty {
+                return completion(nil) //Nothing to update
+            }
+            
+            switch response.responseType {
+            case .partialUpdate:
+                var count = 0
+                self.backgroundContext.performAndWait {
+                    let request: NSFetchRequest<NSFetchRequestResult> = Threat.fetchRequest()
+                    count = (try? self.backgroundContext.count(for: request)) ?? 0
+                }
+                
+                if count == 0 {
+                    return completion(SafeBrowsingError("Partial Update received for non-existent table"))
+                }
+                
+            case .fullUpdate:
+                if !response.removals.isEmpty {
+                    return completion(SafeBrowsingError("Indices to be removed included in a Full Update"))
+                }
+                
+                self.backgroundContext.performAndWait { [weak self] in
+                    guard let self = self else { return completion(nil) }
+                    let request = { () -> NSBatchDeleteRequest in
+                        let request: NSFetchRequest<NSFetchRequestResult> = Threat.fetchRequest()
+                        request.predicate = NSPredicate(format: "threatType == %@", response.threatType.rawValue)
+                        
+                        let deleteRequest = NSBatchDeleteRequest(fetchRequest: request)
+                        deleteRequest.resultType = .resultTypeObjectIDs
+                        return deleteRequest
+                    }()
+                    
+                    if let result = (try? self.backgroundContext.execute(request)) as? NSBatchDeleteResult {
+                        if let deletedObjects = result.result as? [NSManagedObjectID] {
+                            NSManagedObjectContext.mergeChanges(
+                                fromRemoteContextSave: [NSDeletedObjectsKey: deletedObjects],
+                                into: [self.mainContext, self.backgroundContext]
+                            )
+                        }
+                    } else {
+                        let request: NSFetchRequest<Threat> = Threat.fetchRequest()
+                        request.predicate = NSPredicate(format: "threatType == %@", response.threatType.rawValue)
+                        
+                        (try? self.backgroundContext.fetch(request))?.forEach({
```

Comment:
can remove round brackets around `try` and `forEaach`

Reply:
you can try this
`try? self.backgroundContext.fetch(request).forEach {`
compiles for me